lib/events/firestoreevents/firestoreevents.go

```diff
@@ -500,6 +501,15 @@ func (l *Log) SearchEvents(fromUTC, toUTC time.Time, namespace string, eventType
 		return nil, "", firestorebk.ConvertGRPCError(err)
 	}
 
+	// Correctly detecting if you've reached the end of a query in firestore is
+	// tricky since it doesn't set any flag when it finds that there are no further events.
+	// This solution here seems to be the the most common but I haven't been able to find
+	// any documented hard guarantees on firestore not early returning for some reason like response
```

Comment:
Took a peek at the source code of `firestore.DocumentItarator`.  Looks like `GetAll()` is just a convenience wrapper around `Next()`, which collects all results until `iterator.Done` reached.  Since the docs guarantee that `Next()` only returns `iterator.Done` if there are actually no more documents matching the query, I think we can treat this as *effectively* guaranteed.